### PR TITLE
[CINFRA] Replication2 adaptation of shell-transaction-intermediate-commit-cluster

### PR DESCRIPTION
**Pull request by apetenchea**
### Scope & Purpose

This PR provides the necessary adaptations for the `shell-transaction-intermediate-commit-cluster.js` suite, making it run-able with replication2 default.

Most of these changes revolve around the following two differences:
- In replication2 we do not "drop" followers
- In replication2, followers do not perform intermediate commits by themselves, unless the leader replicates an `IntermediateCommit` operation. This means that if intermediate commits are disabled on the leader, they are inherently disabled on the follower.

Although they were initially targeted at replication1-specific behavior, I believe the suite is still relevant for replication2, at least from the perspective of a more general sanity check: making sure we don't crash, and the leader and follower are kept in sync.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### CODE REVIEW

File: tests/js/client/shell/transaction/shell-transaction-intermediate-commit-cluster.js [hunk 1]
```diff
@@ -165,14 +173,22 @@ function transactionIntermediateCommitsSingleSuite() {
...
-      
-      // follower must have been dropped
-      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
-      assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+
+      if (!isReplication2) {
+        // follower must have been dropped
+        let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
+        assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+      }
```
Comment (mchacki) [inline tests/js/client/shell/transaction/shell-transaction-intermediate-commit-cluster.js:181]: Should we add the "else" branch, that asserts that droppedFollowers is not changed?

The metric is simply not in use in Rep2 as the concept does not exist.

File: tests/js/client/shell/transaction/shell-transaction-intermediate-commit-cluster.js [hunk 2]
```diff
@@ -261,13 +283,22 @@ function transactionIntermediateCommitsSingleSuite() {
       assertEqual(didWork, false);
 
       // follower must have been dropped
-      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
-      assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+      if (!isReplication2) {
+        let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
+        assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+      }
```
Comment (mchacki) [inline tests/js/client/shell/transaction/shell-transaction-intermediate-commit-cluster.js:289]: Again else case?

File: tests/js/client/shell/transaction/shell-transaction-intermediate-commit-cluster.js [hunk 3]
```diff
@@ -312,12 +343,18 @@ function transactionIntermediateCommitsSingleSuite() {
       trx.abort();
       
       // follower must have been dropped
-      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
-      assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+      if (!isReplication2) {
+        let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_total");
+        assertEqual(droppedFollowersBefore + 1, droppedFollowersAfter);
+      }
```
Comment (mchacki) [inline tests/js/client/shell/transaction/shell-transaction-intermediate-commit-cluster.js:349]: Again else?

**Review by mchacki — APPROVED** [review]
LGTM.

my comments are really optional, the second test file actually covers the cases.

**Comment by apetenchea** [comment]
Suggestions addressed in https://github.com/arangodb/arangodb/pull/20240/commits/5df1bcefc29cc6872cbee2542e17b2ed77b2d80f#diff-cbb9de024c04b9a4ad56dcf5241810530885e3ade5cd8ab52ab7fc7057c38f57

**Pull request merged**